crates/types/src/services/executor.rs

```diff
@@ -310,8 +262,6 @@ pub enum Error {
     Backtrace(Box<Backtrace>),
     #[error("Transaction doesn't match expected result: {transaction_id:#x}")]
     InvalidTransactionOutcome { transaction_id: Bytes32 },
-    #[error("Transaction root is invalid")]
-    InvalidTransactionRoot,
```

Comment:
Are we removing this since the headers are guaranteed to be correct by construction?

Reply:
Yes, because we construct the block during validation, we will hash all fields and compare the `BlockId` of all fields.